### PR TITLE
WT-9822 Create a tool to parse the contents of a block in a WT binary file.

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

**Comment by ddanderson** [comment]
@keitharnoldsmith and @pmacko86, can you please review?  This is the start of a python tool that decodes WT binaries.  It's a work in progress (it only does row store internal and leaf pages!), but it's probably already pretty useful.  The idea is to fill in the functionality blanks as we go.

File: tools/wt_binary_decode.py [hunk 1]
```diff
@@ -0,0 +1,652 @@
...
+    p.rint('  disk_size: ' + str(blockhead.disk_size))
+    p.rint('  checksum: ' + hex(uint32(b)))
+    blockhead.flags = uint8(b)
+    p.rint('  block flags: ' + hex(blockhead.flags))
+    if uint8(b) != 0 or uint8(b) != 0 or uint8(b) != 0:
+        p.rint('garbage in unused bytes')
+        return
+
+    if pagehead.type == WT_PAGE_INVALID:
+        pass    # a blank page: TODO mayber should check that it's all zeros?
```
Comment (pmacko86) [inline tools/wt_binary_decode.py:463]: Please fix the spelling

**Review by pmacko86 — APPROVED** [review]
LGTM. It looks like this could be a very useful tool!

File: tools/wt_binary_decode.py [hunk 2]
```diff
@@ -0,0 +1,652 @@
...
+POS_1BYTE_MARKER = 0x80
+POS_2BYTE_MARKER = 0xc0
+POS_MULTI_MARKER = 0xe0
+
+NEG_1BYTE_MIN = -2**6
+NEG_2BYTE_MIN = -2**13 + NEG_1BYTE_MIN
+POS_1BYTE_MAX = 2**6 - 1
+POS_2BYTE_MAX = 2**13 + POS_1BYTE_MAX
+
+MINUS_BIT = -1 << 64
```
Comment (pmacko86) [inline tools/wt_binary_decode.py:79]: I don't see this constant being used. Are you sure it's supposed to be shifted by 64 bits and not 63?

**Review by pmacko86 — APPROVED** [review]

**Review by keitharnoldsmith — APPROVED** [review]
LGTM. I've already found a use for it!

I did spot one typo in a comment.

**Pull request merged**